design.md

```diff
@@ -69,23 +88,27 @@ The client must periodically call a keepalive function in the kernel. If the ker
 
 - There is a file runner interface which specifies:
   - A method to run a script in a repo that accepts a list of objects and returns an object.
+    - The kernel must internally have a method to run a script with a list of devs to implement the gRPC API call to run scripts.
   - A method to load a robot from a robot config file specified as a file specifier and a connection method.
     - A connection method can be a device connection method (e.g. UDP or HID) or a simulator.
-- A script is given a string containing a JSON object. The schema of this object is:
+- When a script is run it is given a string containing a JSON object. The schema of this object is:
   - A dictionary of script environment variables. This is not to be confused with the operating system's environment variables or the JVM's environment variables.
```

Comment:
Is there a reason why this is a string containing a JSON object and not a data structure containing the relevant information? I am assuming that there are only two ways to start a script:
- gRPC call, in which case we are already sending structured data over as a protobuf message
- API call from another script, in which case we get to define the API as we wish

Both cases support a more structured (and less fragile) API.

Reply:
You are right, this is unnecessarily underdefined.